src/rpc/methods/f3.rs

```diff
@@ -385,3 +393,164 @@ impl RpcMethod<1> for GetPowerTable {
         Ok(power_entries)
     }
 }
+
+pub enum ProtectPeer {}
+impl RpcMethod<1> for ProtectPeer {
+    const NAME: &'static str = "F3.ProtectPeer";
+    const PARAM_NAMES: [&'static str; 1] = ["peer_id"];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (String,);
+    type Ok = bool;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore>,
+        (peer_id,): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let peer_id = PeerId::from_str(&peer_id)?;
+        let (tx, rx) = flume::bounded(1);
+        ctx.network_send
+            .send_async(NetworkMessage::JSONRPCRequest {
+                method: NetRPCMethods::ProtectPeer(tx, peer_id),
+            })
+            .await?;
+        rx.recv_async().await?;
+        Ok(true)
+    }
+}
+
+pub enum GetParticipatedMinerIDs {}
+impl RpcMethod<0> for GetParticipatedMinerIDs {
+    const NAME: &'static str = "F3.GetParticipatedMinerIDs";
+    const PARAM_NAMES: [&'static str; 0] = [];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = ();
+    type Ok = Vec<u64>;
+
+    async fn handle(ctx: Ctx<impl Blockstore>, _: Self::Params) -> Result<Self::Ok, ServerError> {
+        match ctx.chain_config().network {
+            NetworkChain::Devnet(_) => {
+                // For now, just hard code the devnet miner for testing
+                let shared_miner_addr = Address::from_str("t01000")?;
+                Ok(vec![shared_miner_addr.id()?])
```

Comment:
It would be worth documenting somewhere the "temporary" solutions. We could make it less temporary via env variable, e.g., `FOREST_DEVNET_SHARED_MINER_ADDRESS`

Reply:
I plan to implement `Filecoin.F3Participate` right after this is merged so it won't live long